gpu_bdb/bdb_tools/q28_utils.py

```diff
@@ -97,22 +131,39 @@ def build_features(t):
 
 def build_labels(reviews_df):
     y = reviews_df["pr_review_rating"].map_partitions(map_labels)
-    y = y.map_partitions(lambda x: cupy.asarray(x, cupy.int32)).persist()
+
+    if isinstance(reviews_df, dask_cudf.DataFrame):
+        y = y.map_partitions(lambda x: cp.asarray(x, np.int32)).persist()
+        y._meta = cp.array(y._meta)
```

Comment:
Why do we need to do this ?

Reply:
Basically the `map_partitions` call creates a dask array with a cupy chunktype but for some reason the metadata shows a numpy chunktype. This issue has a deeper explanation: https://github.com/rapidsai/cudf/issues/4309